pkg/capi/cluster.go

```diff
@@ -0,0 +1,308 @@
+			Version: version,
+			Group:   "bootstrap.cluster.x-k8s.io",
+			Kind:    "TalosConfig",
+		},
+	)
+
+	clusterRef := types.NamespacedName{Namespace: "default", Name: clusterName}
+
+	if err := clusterAPI.runtimeClient.Get(ctx, clusterRef, &cluster); err != nil {
+		return nil, err
+	}
+
+	var (
+		val interface{}
+		err error
+	)
+
+	controlPlaneRef, err := getNamespacedName(cluster, "spec", "controlPlaneRef")
+	if err != nil {
+		return nil, err
+	}
+
+	if err = clusterAPI.runtimeClient.Get(ctx, controlPlaneRef, &controlPlane); err != nil {
+		return nil, err
+	}
+
+	if val, err = getField(controlPlane, "status", "selector"); err != nil {
+		return nil, err
+	}
+
+	labelSelector, err := labels.Parse(val.(string))
+	if err != nil {
+		return nil, err
+	}
+
+	if err = clusterAPI.runtimeClient.List(ctx, &machines, runtimeclient.MatchingLabelsSelector{Selector: labelSelector}); err != nil {
+		return nil, err
+	}
+
+	if len(machines.Items) < 1 {
+		return nil, fmt.Errorf("not enough machines found")
+	}
+
+	configRef, err := getNamespacedName(machines.Items[0], "spec", "bootstrap", "configRef")
+	if err != nil {
+		return nil, err
+	}
+
+	if err = clusterAPI.runtimeClient.Get(ctx, configRef, &talosConfig); err != nil {
+		return nil, err
+	}
+
+	var (
+		clientConfig      *clientconfig.Config
+		ok                bool
+		talosConfigString string
+	)
+
+	val, err = getField(talosConfig, "status", "talosConfig")
+	if err != nil {
+		val, err = getField(talosConfig, "status", "dataSecretName")
+		if err != nil {
+			return nil, err
+		}
+
+		var (
+			secret *v1.Secret
+			raw    []byte
+		)
+
+		secret, err = clusterAPI.clientset.CoreV1().Secrets("default").Get(ctx, val.(string), metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		raw, ok = secret.Data["value"]
+		if !ok {
+			return nil, fmt.Errorf("failed to get talosconfig secret value")
+		}
+
+		talosConfigString = string(raw)
+	} else {
+		talosConfigString, ok = val.(string)
+		if !ok {
+			return nil, fmt.Errorf("failed to convert talosConfig to string")
+		}
+	}
+
+	clientConfig, err = clientconfig.FromString(talosConfigString)
+
+	if err != nil {
+		return nil, err
+	}
+
+	kubeconfig, err := clusterAPI.GetKubeconfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	options := client.GetKubeconfigOptions{
+		Kubeconfig:          kubeconfig,
+		WorkloadClusterName: clusterRef.Name,
+		Namespace:           clusterRef.Namespace,
+	}
+
+	raw, err := clusterAPI.client.GetKubeconfig(options)
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := clientcmd.RESTConfigFromKubeConfig([]byte(raw))
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	nodes, err := clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, node := range nodes.Items {
+		for _, address := range node.Status.Addresses {
+			if address.Type != v1.NodeExternalIP {
```

Comment:
Add all internal IPs to `cp/worker` endpoints. 
ExternalIP to controlPlanes for talosconfig.